### PR TITLE
Updated commons-collections to 4.4 (from 3.2.2)

### PR DESCRIPTION
**Pull request by FSchumacher**
Bugzilla Id: 63809

## Description
Update commons-collection 3.2.2 to commons-collection4 4.4

## Motivation and Context
commons-collection4 is the new mainline of commons-collection and it supports generics.

## How Has This Been Tested?
Run the test suite

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the [code style][style-guide] of this project.
- [ ] I have updated the documentation accordingly.

[style-guide]: https://wiki.apache.org/jmeter/CodeStyleGuidelines

### CODE REVIEW

**Comment by pmouawad** [comment]
Thanks for PR, it looks good to me.
Do we delete commons-collections 3 from lib folder ?
If so (not sure as there are probably dependencies that rely on it) , we should add a breaking change in changes.xml

Thanks

File: src/components/src/main/java/org/apache/jmeter/visualizers/ViewResultsFullVisualizer.java [hunk 1]
```diff
@@ -135,9 +135,9 @@ public ViewResultsFullVisualizer() {
         super();
         final int maxResults = JMeterUtils.getPropDefault("view.results.tree.max_results", 500);
         if (maxResults > 0) {
-            buffer = new CircularFifoBuffer(maxResults);
+            buffer = new CircularFifoQueue<>(maxResults);
         } else {
-            buffer = new UnboundedFifoBuffer();
+            buffer = new LinkedList<>();
```
Comment (vlsi) [inline src/components/src/main/java/org/apache/jmeter/visualizers/ViewResultsFullVisualizer.java]: Could we please stop using `LinkedList`?
Here's what Joshua Bloch (`LinkedList` author) says: https://twitter.com/joshbloch/status/583813919019573248

**Comment by pmouawad** [comment]
Hi Felix,
Should we merge this before releasing ?
One note, I guess commons-collections 3.x is used by many plugins so we might break them.
This will require an explicit mention in release notes.
Regards

File: src/components/src/main/java/org/apache/jmeter/visualizers/ViewResultsFullVisualizer.java (hunk 1 above)
Comment (FSchumacher) [reply src/components/src/main/java/org/apache/jmeter/visualizers/ViewResultsFullVisualizer.java:140]: What is your preferred List here?

**Comment by FSchumacher** [comment]
> Thanks for PR, it looks good to me.
> Do we delete commons-collections 3 from lib folder ?
> If so (not sure as there are probably dependencies that rely on it) , we should add a breaking change in changes.xml
> 
> Thanks

As many plugins might need commons-collections-3 we should probably keep the old library.
I think it still is nice to our code to commons-collections4.

**Pull request closed without merging**